### PR TITLE
Patterns MVP

### PR DESCRIPTION
**Pull request by gwyneplaine**
## Reboot Pattern Guidance
See [Reboot Polaris patterns vault project](https://vault.shopify.io/gsd/projects/30559)

- New layout primitives `Stack`, `Box`, `Grid`, `Row`.
- Playroom patched to support larger code snippets.
- New designs and content for the [patterns](https://polaris.shopify.com/patterns) pages.
- Deprecated old patterns:
  ||Old|New|
  |--|--|--|
  |**design-patterns** | `/patterns/design-patterns` | `404`|
  |**page-layouts** | `/patterns/page-layouts` | `404`|
  |**error-messages** | `/patterns/error-messages` | `/content/error-messages`|
  |**help-content** | `/patterns/help-content` | `/content/help-content`|
  |**loading** | `/patterns/loading` | `/patterns-legacy/loading`|
  |**new-badge** | `/patterns/new-badge` | `/patterns-legacy/new-badge`|
  |**pickers** | `/patterns/pickers` | `/patterns-legacy/pickers`|
  |**text-fields** | `/patterns/text-fields` | `/patterns-legacy/text-fields`|
- Page padding and max-width adjusted such that live previews for components and patterns always display the desktop breakpoint.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: patches/playroom+0.28.0.patch [hunk 1]
```diff
@@ -36,6 +36,174 @@ index 56defa7..1e7cf3b 100644
...
+@@ -32,35 +39,42 @@ export default function Iframe({
+   const intersectionRatio = intersection?.intersectionRatio ?? null;
+ 
+   useEffect(() => {
+-    if (intersectionRatio === null) {
+-      return;
++    if (intersectionRatio === null) return;
++    const listener = (e: any) => {
++      if (e.data.type === READY_TO_RECEIVE && e.source === iframeRef.current?.contentWindow) {
++        console.log(e);
```
Comment (jesstelford) [inline patches/playroom+0.28.0.patch]:
```suggestion
```

File: polaris.shopify.com/content/components/layout-and-structure/page.md [hunk 2]
```diff
@@ -158,3 +159,8 @@ Add a menu item
 - To lay out the content within a page, use the [layout component](https://polaris.shopify.com/components/layout-and-structure/layout)
 - To add pagination within the context of a list or other page content, use the [pagination component](https://polaris.shopify.com/components/navigation/pagination)
 - To add primary and secondary calls to action at the bottom of a page, see the [page actions component](https://polaris.shopify.com/components/actions/page-actions)
+
+## Related patterns
+
+- [App Settings](/patterns/app-settings)
+- [Resource Details](/patterns/resource-details)
```
Comment (jesstelford) [inline polaris.shopify.com/content/components/layout-and-structure/page.md]:
```suggestion
- [App settings layout](/patterns/app-settings-layout)
- [Resource details layout](/patterns/resource-details-layout)
```

File: polaris.shopify.com/content/components/selection-and-input/date-picker.md [hunk 3]
```diff
@@ -57,3 +58,7 @@ If you use the date picker within a [popover component](/components/overlays/pop
 - Press the <kbd>tab</kbd> key to move forward and <kbd>shift</kbd> + <kbd>tab</kbd> to move backward through the previous button, next button, and the calendar
 - When focus is in the calendar, move keyboard focus between the dates using the arrow keys
 - To select a date that has focus, press the <kbd>enter</kbd>/<kbd>return</kbd> key
+
+## Related patterns
+
+- [Date Picking](/patterns/date-picking)
```
Comment (jesstelford) [inline polaris.shopify.com/content/components/selection-and-input/date-picker.md]:
```suggestion
- [Date picking](/patterns/date-picking)
```

File: polaris.shopify.com/src/components/PatternsPage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      <div className={styles.Stack} data-gap="8">
+        <div>
+          <Longform>
+            <h1 className={styles.Header}>Patterns </h1>
+            <p>{description}</p>
+          </Longform>
+          <UpdateBanner
+            message={`We are making changes to the patterns section. Expect changes to
+              happen here and [join the discussion]("https://github.com/Shopify/polaris/discussions/categories/pattern-documentation)
+              to make it better. [See old patterns here]("/patterns-legacy").`}
```
Comment (jesstelford) [inline polaris.shopify.com/src/components/PatternsPage/index.tsx]:
```suggestion
              to make it better. [See old patterns here](/patterns-legacy).`}
```

File: polaris.shopify.com/content/patterns-legacy/new-badge.md [hunk 5]
```diff
@@ -14,7 +18,7 @@ keywords:
 
 There’s opportunity to expand on this documentation. If you’d like to share ideas, or evolve the guidance, join the [Github discussion](https://github.com/Shopify/polaris/discussions/6751) or jump straight to updating the [documentation](https://polaris.shopify.com/contributing/documentation).
 
-![Card with the new badge aligned to the right of the card heading](/images/foundations/patterns/new-badge/new-badge@2x.png)
+![Card with the new badge aligned to the right of the card heading](/images/foundations/patterns-legacy/new-badge/new-badge@2x.png)
```
Comment (jesstelford) [inline polaris.shopify.com/content/patterns-legacy/new-badge.md]:
```suggestion
![Card with the new badge aligned to the right of the card heading](/images/foundations/patterns/new-badge/new-badge@2x.png)
```

File: polaris.shopify.com/content/patterns-legacy/text-fields.md [hunk 6]
```diff
@@ -2,6 +2,9 @@
 title: Text fields
 description: Text fields combine the field label (the title) and the input area. Inputs can vary from typed text to URLs to date pickers.
 icon: FormsMajor
+status:
+  value: Deprecated
+  message: This pattern is no longer supported.
```
Comment (jesstelford) [inline polaris.shopify.com/content/patterns-legacy/text-fields.md:8]:
```suggestion
noIndex: true
status:
  value: Deprecated
  message: This pattern is no longer supported.
```

File: polaris.shopify.com/src/components/PatternThumbnail/PatternThumbnail.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import Image from 'next/image';
+
+interface Props {
+  title: string;
+  img?: string;
+}
+
+function ComponentThumbnail({title, img}: Props) {
```
Comment (jesstelford) [inline polaris.shopify.com/src/components/PatternThumbnail/PatternThumbnail.tsx]:
```suggestion
function PatternThumbnail({title, img}: Props) {
```

File: polaris.shopify.com/src/components/PatternThumbnail/PatternThumbnail.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+          quality={70}
+          sizes="300px"
+          alt={`Screenshot of the ${title} pattern`}
+        />
+      ) : null}
+    </div>
+  );
+}
+
+export default ComponentThumbnail;
```
Comment (jesstelford) [inline polaris.shopify.com/src/components/PatternThumbnail/PatternThumbnail.tsx]:
```suggestion
export default PatternThumbnail;
```

**Review by jesstelford — CHANGES_REQUESTED** [review]
Pairing with @gwyneplaine

File: polaris.shopify.com/src/components/Grid/Grid.tsx [hunk 9]
```diff
@@ -29,40 +68,36 @@ export interface GridItemProps {
...
+      <Box as={as} ref={ref} className={styles.GridItem} {...searchAttributes}>
+        <Link href={url} className={styles.Text}>
+          <SearchResultHighlight />
+          {renderPreview && (
+            <div className={styles.Preview}>{renderPreview()}</div>
+          )}
+          <h4>
+            {title} {status && <StatusBadge status={status} />}
+          </h4>
+          <p>{stripMarkdownLinks(description || '')}</p>
```
Comment (gwyneplaine) [inline polaris.shopify.com/src/components/Grid/Grid.tsx:87]: Why do we need this? Ideally this should not render a p tag if no description exists.

**Review by jesstelford — APPROVED** [review]

**Pull request merged**